### DISCUSSION TITLE
RESTEasy 5.0.0.Final

### DISCUSSION THREAD

**Discussion by jamezp** [resteasy/resteasy#2953] [Announcements]
## Release Notes
        
<h2>        Sub-task
</h2>
<ul>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2979'>RESTEASY-2979</a>] -         remove spring components from resteasy main project
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2987'>RESTEASY-2987</a>] -         Remove resteasy-undertow-spring server adapter
</li>
</ul>
                                        
<h2>        Feature Request
</h2>
<ul>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2887'>RESTEASY-2887</a>] -         JDK17 support
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3031'>RESTEASY-3031</a>] -         allow http client inject through registration / HTTP2 support via vertx engine 
</li>
</ul>
    
<h2>        Bug
</h2>
<ul>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2837'>RESTEASY-2837</a>] -         Improve TS for WF 22.0.1.Final
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2914'>RESTEASY-2914</a>] -         ResteasyViolationException#toString concurrency generate a java.util.ConcurrentModificationException
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2953'>RESTEASY-2953</a>] -         testsuite no longer runs successfully using -Dserver.home
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2966'>RESTEASY-2966</a>] -         resteasy-core is missing from the RESTEasy BOM
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2994'>RESTEASY-2994</a>] -         Fix the search on the generated JavaDoc
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2996'>RESTEASY-2996</a>] -         Reactor netty server adapter mix up response bytes for AsyncOutputStream write 
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2997'>RESTEASY-2997</a>] -         RESTEASY003210 is appeared in response body when request resource is not found
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3000'>RESTEASY-3000</a>] -         Testsuite: Assert.equal should compare actual with expected
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3013'>RESTEASY-3013</a>] -         The Netty 4 Server Adapter test JaxrsAsyncTest is flaky
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3019'>RESTEASY-3019</a>] -         Content-Length header value with leading whitespaces causes Exception
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3022'>RESTEASY-3022</a>] -         Use the correct indicator on whether or not a ContextualExecutor is managed
</li>
</ul>
        
<h2>        Task
</h2>
<ul>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2967'>RESTEASY-2967</a>] -         Remove the Guice module
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2975'>RESTEASY-2975</a>] -         Move the MicroProfile modules to a new project
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2976'>RESTEASY-2976</a>] -         move resteasy-spring components to separate sub-project
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2988'>RESTEASY-2988</a>] -         Migrate the resteasy-jboss-modules to a Galleon Feature Pack
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2993'>RESTEASY-2993</a>] -         Migrate to prefer the Jakarta API dependencies over the JBoss forks
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2999'>RESTEASY-2999</a>] -         Move the Server Cache Maven module to the resteasy-extensions project
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3001'>RESTEASY-3001</a>] -         Remove unneccesary boolean object boxing
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3009'>RESTEASY-3009</a>] -         Create a CI job to test RESTEasy against the latest WildFly release
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3010'>RESTEASY-3010</a>] -         Fix tests that require legacy security
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3014'>RESTEASY-3014</a>] -         Switch CI to use Eclipse Adoptium and update testing to Java 17
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3030'>RESTEASY-3030</a>] -         Replace Java EE references with Jakarta EE references
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3035'>RESTEASY-3035</a>] -         Add documentation for match_cache parameters
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3036'>RESTEASY-3036</a>] -         Add documentation for resteasy.patchfilter.legacy
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3037'>RESTEASY-3037</a>] -         Ignore tests that require modules not provided by WildFly by default
</li>
</ul>
                                                                                                            
<h2>        Component Upgrade
</h2>
<ul>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2991'>RESTEASY-2991</a>] -         Upgrade WildFly Arquillian to 4.0.0.Alpha2
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3008'>RESTEASY-3008</a>] -         Upgrade Galleon Provisioning to 5.2.2.Final
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3041'>RESTEASY-3041</a>] -         Upgrade the wildfly-maven-plugin to 2.1.0.Final
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3042'>RESTEASY-3042</a>] -         Upgrade WildFly to 25.0.1.Final
</li>
</ul>
    
<h2>        Enhancement
</h2>
<ul>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-2880'>RESTEASY-2880</a>] -         Threshold before writing to disk should be configurable
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3004'>RESTEASY-3004</a>] -         Add HTTPS and HTTP2 tests to the resteasy-client-vertx
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3005'>RESTEASY-3005</a>] -         default http port should be set based on http scheme
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3015'>RESTEASY-3015</a>] -         Avoid a payload byte[] copy when using reactor-netty HTTP client engine
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3021'>RESTEASY-3021</a>] -         Create a way to propagate the RESTEasy context for new threads
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3026'>RESTEASY-3026</a>] -         The ThreadContext.reset() should pass the type parameter for resetting
</li>
<li>[<a href='https://issues.redhat.com/browse/RESTEASY-3027'>RESTEASY-3027</a>] -         Allow ThreadContext&#39;s to be enabled via provider
</li>
</ul>
                                                                                                                        

<hr /><em>This discussion was created from the release <a href='https://github.com/resteasy/resteasy/releases/tag/5.0.0.Final'>RESTEasy 5.0.0.Final</a>.</em>

**Comment by vanta** [comment]
Hello, 

I have a question, maybe I don't fully get it, so please explain.

I see ticket [RESTEASY-3030](https://issues.redhat.com/browse/RESTEASY-3030) as being included in `5.x` family but I can still see imports from `javax.enterprise` in `5.0.6.Final` release (see [here](https://github.com/resteasy/resteasy/blob/5.0.6.Final/resteasy-cdi/src/main/java/org/jboss/resteasy/cdi/ResteasyCdiExtension.java)). 

Why is that?

**Reply by jamezp** [reply]
Hello @vanta,
The wording in that JIRA is a bit misleading now that I re-read it. What that JIRA did was rename the "Java EE" references to "Jakarta EE" since we target Jakarta RESTful Web Services 2.1 in 5.0. The package rename was not done until 6.0.0.Final and is now Jakarta RESTful Web Services 3.1 in 6.2.3.Final.

**Reply by vanta** [reply]
Thank you for your quick response.

Let me follow up: so why https://developer.jboss.org/en/resteasy/blog/2019/10/30/resteasy-440final-moving-to-jakarta-artifacts-and-much-more states that:
> In addition to that, RESTEasy 4.4.0.Final is the first version relying directly or indirectly on Jakarta EE API artifacts. 

Is it only for some artifacts, not for all?

**Reply by jamezp** [reply]
Yes, the first change was for artifacts only. The second change was for documentation to remove, as much as we could, the reference to "Java EE" as we are targeting Jakarta EE.
